runtime/src/benchmarking/criterion.rs

```diff
@@ -0,0 +1,147 @@
+/// Criterion benchmark executor implementations.
+use criterion::async_executor::AsyncExecutor;
+use futures::Future;
+use std::any::Any;
+use std::cell::RefCell;
+use std::thread_local;
+
+thread_local! {
+    static CONTEXT: RefCell<Option<Box<dyn Any + Send>>> = RefCell::new(None);
+}
+
+/// Set the context value
+fn set_context<C: Clone + Send + 'static>(context: C) {
+    CONTEXT.with(|cell| {
+        *cell.borrow_mut() = Some(Box::new(context));
+    });
+}
+
+/// Get the context value
+pub fn context<C: Clone + Send + 'static>() -> C {
+    CONTEXT.with(|cell| {
+        let borrow = cell.borrow();
+        match borrow.as_ref() {
+            Some(any) => any
+                .downcast_ref::<C>()
+                .expect("Context type mismatch ‑ internal error")
+                .clone(),
+            None => panic!(
+                "No context of type {} available. Make sure you're using the correct executor.",
+                std::any::type_name::<C>()
+            ),
+        }
+    })
+}
+
+/// Clear the context value
+fn clear_context() {
+    CONTEXT.with(|cell| {
+        *cell.borrow_mut() = None;
+    });
+}
+
+/// Convenience module for tokio-specific executor
+pub mod tokio {
```

Comment:
We should remove this nesting. I wonder if there is a way to still provide a config for this benchmarking?

Reply:
We can create a "benchmarker" struct that always instantiates with the same config.